### DISCUSSION TITLE
【严重问题】文本框模式文字位置显示错误

### DISCUSSION THREAD

**Discussion by mianma12138** [Tencent/libpag#2419] [Q&A] [answered]
[出问题的文件.zip](https://github.com/user-attachments/files/16599206/default.zip)
这个pag测试文件出现两个问题：文本显示位置出错，不从第一行开始展示
![image](https://github.com/user-attachments/assets/00447f17-c996-40fc-9c9e-6882feec4f96)
![image](https://github.com/user-attachments/assets/fe6175da-09bc-44b0-a940-a33c95ea3238)

**Comment by mianma12138** [comment]
这个pag测试文件出现两个问题：
（1）文本显示位置出错，不从第一行开始展示。
（2）整个素材只用了一层bmp合成，单上机展示卡顿跳帧，已经在AE中做了一些优化。

**Comment by mianma12138** [comment]
管理员看下这个帖子，测试文件已经上传。

**Comment by mianma12138** [comment]
![image](https://github.com/user-attachments/assets/b45ecfe3-4ae1-4d20-b09e-466cb32ab04f)

**Comment by mianma12138** [comment]
pag做全屏或者半屏类的动效，上机就会卡顿，有没有优化的方法。看了下官网给的全屏素材也是会卡顿，并不像宣传片中的那样，很流畅的播放全屏动效。官网的案例使用了一个大型赛车动效，看起来很流畅。请问管理员是否有优化的方法。
![image](https://github.com/user-attachments/assets/45003791-e1fa-46a6-9ca4-094e71260151)

**Reply by domchen** [reply]
请注意看文档：https://pag.art/docs/editable-text.html

**Reply by mianma12138** [reply]
管理员你好，文档中的几个方法，我上面帖子都用到了，并没有解决我的问题。当文本框的宽度大于实际文本的宽度时，他不会从第一行开始排版，而是整体居中。文档中没有提到这一点的解决方案。

**Reply by mianma12138** [reply]
“如果只需要换行不需要缩小，则用框文本制作，且将框的宽度设置成你想要的宽度，框的高度设置尽量大一些，这样就不会缩小字号，只会增加行数。”
这个方法也无法解决遇到的问题

**Reply by domchen** [reply, marked as answer]
这里文档好像被覆盖了，我们有时间还原下之前的记录。点文本都是和AE排版完全一致的，框文本排版这块相比AE核心是有两个额外规则：1.框不够大时会缩小字号。2.强制垂直居中。你现在遇到的问题就是第二点。绝大部分使用场景下用户需要的就是垂直居中，所以框高度设置大一些是正好符合需求的。你现在如果是想要「自动换行」而且还要「顶端对齐」，这个用框文本目前没法同时做到。正确的做法是用点文本，自己加换行符\n来手动换行就行了。如果你的初始文本行数是比较固定的，只是想继续增加文本的时候缩小字号。不出现垂直方向位置偏移，这个也可以达到。就是AE里把文本框拉到正好显示满你当前的文本内容，再小一点就有一行显示不出来的状态。这样在PAG里面也是正好对应AE的位置。继续增文本内容只会缩小字号。

**Reply by mianma12138** [reply]
感谢管理员解答，看来只能这样处理了。请问pag开源项目的话，以后会有这方面的优化吗

**Reply by domchen** [reply]
未来计划加个开关允许用户关闭垂直居中功能。但近期都没有人力投入开发。
